src/test.ts

```
@@ -10,157 +10,136 @@ const tscPath = new URL(
-          fs.readFileSync(new URL(file, outputFolder)).toString(),
-        )
-      : null;
+    const baseline = isBaselineFile ? readFileContent(baselinePath) : null;
+    const generated = isOutputFile ? readFileContent(outputPath) : null;
 
     if (baseline !== null || generated !== null) {
       if (baseline !== generated) {
-        console.error(
-          `Test failed: '${file}' is different from baseline file.`,
-        );
+        console.error(`Test failed: '${file}' is different from baseline.`);
         printUnifiedDiff(baseline ?? "", generated ?? "");
         return false;
       }
-
       continue;
     }
 
-    if (baselineStats?.isDirectory() || outputStats?.isDirectory()) {
-      const childBaselineFolder = new URL(`${file}/`, baselineFolder);
-      const childOutputFolder = new URL(`${file}/`, outputFolder);
-      if (!compareToBaselines(childBaselineFolder, childOutputFolder)) {
+    if (fs.existsSync(baselinePath) || fs.existsSync(outputPath)) {
+      if (
+        !compareToBaselines(
+          new URL(`${file}/`, baselineFolder),
+          new URL(`${file}/`, outputFolder),
+        )
+      ) {
         return false;
       }
-
-      continue;
     }
   }
   return true;
 }
 
-function compileGeneratedFiles(lib: string, ...files: string[]) {
+function compileGeneratedFiles(lib: string[] | string, ...files: string[]) {
   try {
+    const fileArgs = files
+      .map((file) => fileURLToPath(new URL(file, outputFolder)))
+      .join(" ");
     child_process.execSync(
-      `node ${fileURLToPath(
-        tscPath,
-      )} --strict --lib ${lib} --types --noEmit ${files
-        .map((file) => fileURLToPath(new URL(file, outputFolder)))
-        .join(" ")}`,
+      `node ${fileURLToPath(tscPath)} --strict --lib ${lib} --types --noEmit ${fileArgs}`,
     );
-  } catch (e: any) {
-    console.error(`Test failed: could not compile '${files.join(",")}':`);
-    console.error(e.stdout.toString());
-    console.error();
```

Skipping this would hide the actual error which would make debugging harder.